tests/jerry/es.next/object-property-redefiniton.js

```diff
@@ -0,0 +1,41 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function checkOwnProperties(obj, propList)
+{
+  names = Object.getOwnPropertyNames(obj)
+  assert(names.length === propList.length)
+
+  for (var i = 0; i < propList.length; ++i)
+  {
+    assert(names[i] === propList[i])
+
+    var descriptor = Object.getOwnPropertyDescriptor(obj, names[i])
+    assert(descriptor.writable == true && descriptor.get === undefined && descriptor.set === undefined);
+  }
+}
+
+var o = {
+  get a() {
+    return 5;
+  },
+  b:6,
+  set c(_) {
+  },
+  d:10,
+  a: 11,
+  c: 12
```

Comment:
What about the other direction? When the property is a data property first and redefined as accessor.

Reply:
I've updated the PR according to it.